File: app/javascript/channels/chat_room_channel.js
import consumer from "./consumer"

document.addEventListener('turbolinks:load', () => {
    const chatRoomElement = document.getElementById('chat_room_id');
    if (chatRoomElement) {
        const chatRoomId = chatRoomElement.getAttribute('data-chat-room-id');

        consumer.subscriptions.create({ channel: "ChatRoomChannel", chat_room_id: chatRoomId }, {
            connected() {
                console.log(`Connected to ChatRoomChannel ${chatRoomId}`);
            },

            disconnected() {
                // Called when the subscription has been terminated by the server
            },

            received(data) {
                const messages = document.getElementById('messages');
                messages.insertAdjacentHTML('beforeend', data.message);
            }
        });
    }
});

File: app/channels/chat_room_channel.rb
class ChatRoomChannel < ApplicationCable::Channel
  def subscribed
    stream_from "chat_room_#{params['chat_room_id']}_channel"
  end

  def unsubscribed
    # Any cleanup needed when channel is unsubscribed
  end

  def send_message(data)
    current_user.messages.create!(content: data['message'], chat_room_id: data['chat_room_id'])
  end
end
# class ChatRoomChannel < ApplicationCable::Channel
#   def subscribed
#     stream_from "chat_room_#{params['chat_room_id']}_channel"
#   end
#
#   def unsubscribed
#     # Any cleanup needed when channel is unsubscribed
#   end
#
#   def send_message(data)
#     current_user.private_messages.create!(content: data['message'], chat_room_id: data['chat_room_id'])
#   end
# end
